src/transformers/modeling_utils.py

```diff
@@ -4430,7 +4440,15 @@ def _find_mismatched_keys(
                 # Skip the load for shards that only contain disk-offloaded weights when using safetensors for the offload.
                 if shard_file in disk_only_shard_files:
                     continue
-                state_dict = load_state_dict(shard_file, is_quantized=is_quantized)
+                map_location = None
+                if (
+                    device_map is not None
+                    and "" in device_map
+                    and is_quantized
+                    and isinstance(hf_quantizer, TorchAoHfQuantizer)
+                ):
+                    map_location = torch.device(device_map[""])
```

Comment:
WDYT ? This way, it will also work for multi-gpu case since device_map will be a dict + check that we are in `int4_weight_only` case 
```suggestion
                if (
                    device_map is not None
                    hf_quantizer is not None
                    and hf_quantizer.quantization_config.quant_method == QuantizationMethod.TORCHAO
                    and hf_quantizer.quant_type = "int4_weight_only"
                ):
                    map_location = torch.device([d for d in device_map.values() if d not in ["cpu", "disk"]][0])
```